File: back/models/poll.go
package models

import (
	"gorm.io/gorm"
)

type Poll struct {
	gorm.Model
	Question   string              `gorm:"not null" json:"question" validate:"required,min=3,max=255"`
	IsClosed   bool                `gorm:"default:false" json:"is_closed" validate:"boolean"`
	IsMultiple bool                `gorm:"default:false" json:"is_multiple" validate:"boolean"`
	GroupID    uint                `gorm:"not null" json:"group_id" validate:"required"`
	Group      *Group              `json:"group,omitempty"`
	UserID     uint                `gorm:"not null" json:"user_id"`
	User       *User               `json:"user,omitempty"`
	EventID    *uint               `gorm:"default:null" json:"event_id" validate:"omitempty"`
	Event      *Event              `json:"event,omitempty"`
	Choices    *[]PollAnswerChoice `json:"choices,omitempty" validate:"required,min=1,dive,required"`
}

type PollCreateOrEdit struct {
	Question   *string                         `json:"question" validate:"required,min=3,max=255"`
	IsClosed   *bool                           `json:"is_closed" validate:"omitempty,boolean"`
	IsMultiple *bool                           `json:"is_multiple" validate:"omitempty,boolean"`
	GroupID    *uint                           `json:"group_id" validate:"required"`
	EventID    *uint                           `json:"event_id" validate:"omitempty"`
	Choices    *[]PollAnswerChoiceCreateOrEdit `json:"choices,omitempty" validate:"required,min=1,dive,required"`
}

func (p *PollCreateOrEdit) ToPoll(userId uint) Poll {
	var choices []PollAnswerChoice
	for _, choice := range *p.Choices {
		choices = append(choices, choice.ToPollAnswerChoice())
	}

	var isClosed = false
	if p.IsClosed != nil {
		isClosed = *p.IsClosed
	}

	var isMultiple = false
	if p.IsMultiple != nil {
		isMultiple = *p.IsMultiple
	}

	return Poll{
		Question:   *p.Question,
		IsClosed:   isClosed,
		IsMultiple: isMultiple,
		GroupID:    *p.GroupID,
		EventID:    p.EventID,
		UserID:     userId,
		Choices:    &choices,
	}
}

File: front/lib/core/models/poll.dart

import 'package:front/core/models/poll_choice.dart';

class Poll {
  final int id;
  final String question;
  final bool isClosed;
  final bool isMultiple;
  final int groupId;
  final int userId;
  final int? eventId;
  final List<PollChoice>? choices;

  Poll({
    required this.id,
    required this.question,
    required this.isClosed,
    required this.isMultiple,
    required this.groupId,
    required this.userId,
    this.eventId,
    required this.choices,
  });

  factory Poll.fromJson(Map<String, dynamic> json) {
    return Poll(
      id: json['ID'],
      question: json['question'],
      isClosed: json['is_closed'],
      isMultiple: json['is_multiple'],
      groupId: json['group_id'],
      userId: json['user_id'],
      eventId: json['event_id'],
      choices: (json['choices'] as List).map((choice) => PollChoice.fromJson(choice)).toList(),
    );
  }

  PollCreateOrEdit toCreateOrEdit() {
    return PollCreateOrEdit(
      id: id,
      question: question,
      isMultiple: isMultiple,
      isClosed: isClosed,
      choices: choices?.map((choice) => choice.toCreateOrEdit()).toList(),
      groupId: groupId,
      eventId: eventId,
    );
  }
}

class PollCreateOrEdit {
  final int? id;
  final String? question;
  final bool? isMultiple;
  final bool? isClosed;
  final List<PollChoiceCreateOrEdit>? choices;
  final int? groupId;
  final int? eventId;

  PollCreateOrEdit({
    this.id,
    this.question,
    this.isMultiple,
    this.isClosed,
    this.choices,
    this.groupId,
    this.eventId,
  });

  Map<String, dynamic> toJson() {
    Map<String, dynamic> data = {
      'question': question,
      'is_multiple': isMultiple ?? false,
      'is_closed': isClosed ?? false,
      'choices': choices?.map((choice) => choice.toJson()).toList(),
    };

    if (groupId != null) {
      data['group_id'] = groupId;
    }

    if (eventId != null) {
      data['event_id'] = eventId;
    }

    if (id != null) {
      data['ID'] = id;
    }

    return data;
  }
}